### DISCUSSION TITLE
Self-hosted runner

### DISCUSSION THREAD

**Discussion by andy5995** [KaruroChori/vs-fltk#55] [General]
@KaruroChori You might be interested in [a self-hosted runner](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) sometime. They are pretty simple to set up. I set one up once in a docker container. No port-forwarding or visibility to the outside world is necessary, iirc.

**Comment by KaruroChori** [comment]
I use self-hosted runners for my private gittea instance, but I never checked how to set them up for github.  

Reading the docs, unless I virtualize the runner on systems without too many exploits, it is only suggested to run them for private repositories, or I might get my server/network compromised.
Still, it would be nice so to have a native linux environment on arm, but riscv would still not be supported by github :(.

I am moving this to an issue and I will review it few weeks from now.
